akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -70,8 +72,8 @@ class HttpExt private[http] (private val config: Config)(implicit val system: Ex
 
   private[this] final val DefaultPortForProtocol = -1 // any negative value
 
-  private type ServerLayerBidiFlow = BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed]
-  private type ServerLayerFlow = Flow[ByteString, ByteString, Future[Done]]
+  private type ServerLayerBidiFlow = BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, ServerTerminator]
+  private type ServerLayerFlow = Flow[ByteString, ByteString, (Future[Done], ServerTerminator)]
```

Comment:
the type is never exposed completely to users, so i'm not worried about changing it